hbase-client/src/main/java/org/apache/hadoop/hbase/security/AbstractHBaseSaslRpcClient.java

```diff
@@ -88,6 +88,14 @@ protected AbstractHBaseSaslRpcClient(Configuration conf,
     }
   }
 
+  /**
+   * Computes the initial response a client sends to a server to begin the SASL
+   * challenge/response handshake. If the client's SASL mechanism does not require
+   * that an initial response is sent to begin the handshake, this method will return
+   * an empty byte array, indicating no initial response needs to be sent by this client.
+   *
+   * @return The client's initial response to send the server, or an empty byte array
+   */
   public byte[] getInitialResponse() throws SaslException {
```

Comment:
I think we better returns null for no initial response here? Maybe some sasl implementation just use an empty token...

Reply:
All of the mechanisms in openjdk do have a non-empty token, but I agree with your opinion. Fixed this.